core/src/stored/backends/crud_storage.cc

```diff
@@ -0,0 +1,323 @@
+#include "include/bareos.h"
+#include <fmt/format.h>
+#include "crud_storage.h"
+#include "lib/berrno.h"
+#include "lib/bpipe.h"
+#include "lib/bstringlist.h"
+#include "stored/stored_conf.h"
+#include "stored/stored_globals.h"
+#include <sys/stat.h>
+#include <cctype>
+
+namespace {
+class BPipeHandle {
+  Bpipe* bpipe;
```

Comment:
```suggestion
  Bpipe* bpipe{nullptr};
```
otherwise the assertions will not always work correctly.